README.md

```diff
@@ -1,98 +1,206 @@
+experience. We would suggest using Orbit 2 over Orbit 1 for new and existing
+projects as we are actively working on it.
 
-We drew inspiration from [Managing State with RxJava by Jake Wharton](https://www.reddit.com/r/androiddev/comments/656ter/managing_state_with_rxjava_by_jake_wharton/),
-[RxFeedback](https://github.com/NoTests/RxFeedback.kt) and [Mosby MVI](https://github.com/sockeqwe/mosby).
+If you are risk-averse, we will continue to support Orbit 1 with
+necessary fixes for some time as we use it in our projects.
 
-For more details about MVI and our implementation, please read
+[Orbit 1 is available here.](https://github.com/babylonhealth/orbit-mvi/tree/orbit/main)
 
-1. [MVI overview](docs/overview.md).
-1. [Creating flows in Orbit](docs/orbits.md).
+The readme below covers only Orbit 2.
 
 ## Getting started
 
-[![Download](https://api.bintray.com/packages/babylonpartners/maven/orbit/images/download.svg)](https://bintray.com/babylonpartners/maven/orbit/_latestVersion)
+Orbit 2 is a modular framework. Only one module is required with additional
+functionality provided through plugins included in separate modules. Install
+only what you need!
 
-Include the following dependencies in your build.gradle.kts file:
+At the very least you will need the `orbit-core` module to get started.
 
 ```kotlin
-implementation("com.babylon.orbit:orbit:<latest-version>")
-implementation("com.babylon.orbit:orbit-android:<latest-version>")
+implementation("com.babylon.orbit2:orbit-core:<latest-version>") // <-- This module is mandatory
+implementation("com.babylon.orbit2:orbit-coroutines:<latest-version>")
+implementation("com.babylon.orbit2:orbit-rxjava2:<latest-version>")
+implementation("com.babylon.orbit2:orbit-livedata:<latest-version>")
+implementation("com.babylon.orbit2:orbit-savedstate:<latest-version>")
+
+testImplementation("com.babylon.orbit2:orbit-test:<latest-version>")
 ```
 
-A real-world redux system might look as follows:
+[![Download](https://api.bintray.com/packages/babylonpartners/maven/orbit-core/images/download.svg)](https://bintray.com/babylonpartners/maven/orbit-core/_latestVersion)
 
-``` kotlin
-data class State(val total: Int = 0)
+For detailed documentation, see:
 
-data class AddAction(val number: Int)
+- [Core module and architecture overview](orbit-2-core/README.MD)
+- [Coroutines](orbit-2-coroutines/README.md)
+- [RxJava 2](orbit-2-rxjava2/README.md)
+- [LiveData](orbit-2-livedata/README.md)
+- [Saved state](orbit-2-savedstate/README.md)
+- [Test](orbit-2-test/README.md)
 
-sealed class SideEffect {
-    data class Toast(val text: String) : SideEffect()
+### A word on DSL plugins
+
+Modules like `orbit-coroutines` or `orbit-rxjava2` contain plugins that extend
+Orbit's DSL. In order for Orbit to recognise these plugins, they need to be
+installed at runtime, like so:
+
+``` kotlin
+class MyApplication : Application() {
+    override fun onCreate() {
+        OrbitDslPlugins.register(
+            CoroutineDslPlugin,
```

Comment:
The more I see this the more I think the library should just handle this under the hood... happy if this is directly reflective and we'd need to make sure consumer ProGuard rule exists to keep the plugins.

I mention it also though as it makes the documentation somewhat more complex... we end up needing to duplicate this documentation (already duplicated at `orbit-2-core/README.md:97`), but probably also in the documentation for the different plugins.

Of course if it happened automatically things become far simpler from the users perspective.

Reply:
Rewrote to automatically register the plugins as needed